### PR TITLE
FT-820 Add SnowflakeDynamicTable as possible parent-type.

### PR DESCRIPTION
**Pull request by ErnestoLoma**

### CODE REVIEW

File: pyatlan/model/assets/core/column.py [hunk 1]
```diff
@@ -1574,7 +1574,7 @@ def create(
                 database_qualified_name=database_qualified_name,
             )
 
-            if parent_type == Table:
+            if parent_type in {Table, SnowflakeDynamicTable}:
                 column.table_qualified_name = parent_qualified_name
                 column.table = Table.ref_by_qualified_name(parent_qualified_name)
                 column.table_name = parent_name
```
Comment (Aryamanz29) [inline pyatlan/model/assets/core/column.py:1580]: I have a doubt here. I noticed that the column has the `attribute snowflake_dynamic_table`. Shouldn't we use that here 🤔 

```suggestion
            elif parent_type == SnowflakeDynamicTable:
                column.table_qualified_name = parent_qualified_name
                column.snowflake_dynamic_table = SnowflakeDynamicTable.ref_by_qualified_name(
                    parent_qualified_name
                )
                column.table_name = parent_name
```

This is similar to what we do for other `parent_type` values, such as `MaterializedView`:

```py
     elif parent_type == MaterialisedView:
                column.view_qualified_name = parent_qualified_name
                column.materialised_view = MaterialisedView.ref_by_qualified_name(
                    parent_qualified_name
                )
                column.view_name = parent_name
  ```

File: tests/unit/model/column_test.py [hunk 2]
```diff
@@ -97,6 +103,28 @@ def test_create_when_parent_is_table():
...
+    assert sut.qualified_name == TABLE_COLUMN_QUALIFIED_NAME
+    assert sut.connector_name == CONNECTOR_TYPE
+    assert sut.schema_name == SCHEMA_NAME
+    assert sut.schema_qualified_name == SCHEMA_QUALIFIED_NAME
+    assert sut.database_name == DATABASE_NAME
+    assert sut.database_qualified_name == DATABASE_QUALIFIED_NAME
+    assert sut.connection_qualified_name == CONNECTION_QUALIFIED_NAME
+    assert sut.order == 1
+    assert sut.table_qualified_name == TABLE_QUALIFIED_NAME
+    assert sut.table.qualified_name == TABLE_QUALIFIED_NAME
```
Comment (Aryamanz29) [inline tests/unit/model/column_test.py:124]: Same test assertion is listed twice here ^

Comment (ErnestoLoma) [reply tests/unit/model/column_test.py]: Fixed.

**Review by Aryamanz29 — APPROVED** [review]
LGTM 🚀

**Pull request merged**